Cython/Compiler/Optimize.py

```diff
@@ -228,6 +228,12 @@ def _optimise_for_loop(self, node, iterable, reversed=False):
             return self._transform_bytes_iteration(node, iterable, reversed=reversed)
         if iterable.type is Builtin.unicode_type:
             return self._transform_unicode_iteration(node, iterable, reversed=reversed)
+        # in principle _transform_indexable_iteration would work on most of the above, and
+        # also tuple and list. However, it probably isn't quite as optimized
+        if iterable.type is Builtin.bytearray_type:
+            return self._transform_indexable_iteration(node, iterable, reversed=reversed)
```

Comment:
Is this actually tested anywhere? Including changes to the bytearray while iterating over it? This works in Python, for example:
```python
>>> b = bytearray(b'abcdefg')
>>> b
bytearray(b'abcdefg')
>>> for x in b:
...   print(x, b)
...   del b[-1]
97 bytearray(b'abcdefg')
98 bytearray(b'abcdef')
99 bytearray(b'abcde')
100 bytearray(b'abcd')
>>> b = bytearray(b'abcdefg')
>>> for x in b:
...   print(chr(x), b)
...   del b[0]
a bytearray(b'abcdefg')
c bytearray(b'bcdefg')
e bytearray(b'cdefg')
g bytearray(b'defg')

```

Reply:
I think I did write originally some simple tests and then failed to add it to git (which is obviously no use all at...)

It's now tested properly, including modifying it.